modin/dataframe/io.py

```diff
@@ -83,51 +85,127 @@ def _split_df(pandas_df, chunksize):
+        if not skip and callable(skiprows):
+            skip |= skiprows(lines_read)
+        elif not skip and hasattr(skiprows, "__contains__"):
+            skip |= lines_read in skiprows
+
+        if not skip:
+            header_lines_skipped += 1
+            if header_lines_skipped == header_lines:
+                return lines_read
+
+    return lines_read
+
+
+def _read_csv_from_file(filepath, npartitions, kwargs={}):
+    """Constructs a DataFrame from a CSV file.
 
+    Args:
+        filepath (str): path to the CSV file.
+        npartitions (int): number of partitions for the DataFrame.
+        kwargs (dict): args excluding filepath provided to read_csv.
 
-def _infer_column(first_line, kwargs={}):
-    return pandas.read_csv(BytesIO(first_line), **kwargs).columns
+    Returns:
+        DataFrame or Series constructed from CSV file.
+    """
+    empty_pd_df = pandas.read_csv(
+            filepath, **dict(kwargs, nrows=0, skipfooter=0, skip_footer=0))
+    names = empty_pd_df.columns
+
+    skipfooter = kwargs["skipfooter"]
+    skip_footer = kwargs["skip_footer"]
+
+    partition_kwargs = dict(
+            kwargs, header=None, names=names, skipfooter=0, skip_footer=0)
+    with open(filepath, "rb") as f:
+        # Get the BOM if necessary
+        prefix = b""
+        if kwargs["encoding"] is not None:
+            prefix = f.readline()
+            partition_kwargs["skiprows"] = 1
+            f.seek(0, os.SEEK_SET)  # Return to beginning of file
+
+        prefix_id = ray.put(prefix)
+        partition_kwargs_id = ray.put(partition_kwargs)
+
+        # Skip the header since we already have the header information
+        _skip_header(f, kwargs)
+
+        # Launch tasks to read partitions
+        partition_ids = []
+        index_ids = []
+        total_bytes = os.path.getsize(filepath)
+        chunk_size = max(1, (total_bytes - f.tell()) // npartitions)
+        while f.tell() < total_bytes:
+            start = f.tell()
+            f.seek(chunk_size, os.SEEK_CUR)
+            f.readline()    # Read a whole number of lines
+
+            if f.tell() >= total_bytes:
+                kwargs["skipfooter"] = skipfooter
+                kwargs["skip_footer"] = skip_footer
+
+            partition_id, index_id = _read_csv_with_offset._submit(
+                    args=(filepath, start, f.tell(), partition_kwargs_id,
+                          prefix_id),
+                    num_return_vals=2)
+            partition_ids.append(partition_id)
+            index_ids.append(index_id)
+
+        # Construct index
+        index_id = get_index.remote([empty_pd_df.index.name], *index_ids) \
+            if kwargs["index_col"] is not None else None
+
+        df = DataFrame(
+                row_partitions=partition_ids, columns=names, index=index_id)
+
+        skipfooter = kwargs["skipfooter"] or kwargs["skip_footer"]
+        if skipfooter:
+            df = df.drop(df.index[-skipfooter:])
```

Comment:
Performs badly when `skipfooter` is large. I think we can assume that `skipfooter` is small in most cases.